src/httpc.c

```diff
@@ -464,6 +464,36 @@ httpc_set_accept_encoding(struct httpc_request *req, const char *encoding)
 #endif
 }
 
+int
+httpc_set_http_version(struct httpc_request *req, const char *version)
+{
+	if (strcmp(version, "1.0") == 0){
+		curl_easy_setopt(req->curl_request.easy, 
+			CURL_HTTP_VERSION_1_0, version);
+	}
+	if (strcmp(version, "1.1") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_1_1, version);
+        }
+	if (strcmp(version, "2") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2, version);
+        }
+	if (strcmp(version, "2-tls") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2TLS, version);
+        }
+	if (strcmp(version, "2-prior-knowledge") == 0){
+                curl_easy_setopt(req->curl_request.easy,
+                        CURL_HTTP_VERSION_2_PRIOR_KNOWLEDGE, version);
+        }
+	else{
+		diag_set(IllegalParams, "Not a supported http version");
+		return -1;
+	}
```

Comment:
AFAIU, this else-branch is called on any value that is not equal to `2-prior-knowledge`. For a valid one too: for example, for `1.1`.

Reply:
IOW, we need just do the if-else if-else if-else chain, not separate if statements.

Reply:
Fixed by using else-if chain